awx/main/management/commands/register_default_execution_environments.py

```diff
@@ -68,35 +68,54 @@ def handle(self, *args, **options):
                 name="Default Execution Environment Registry Credential", managed_by_tower=True, credential_type=registry_cred_type[0]
             )
 
-            if created:
-                print("Default Execution Environment Credential registered.")
+            if cred_created:
+                print("'Default Execution Environment Credential' registered.")
 
             inputs = {
                 "host": options.get("registry_url"),
                 "password": options.get("registry_password"),
                 "username": options.get("registry_username"),
                 "verify_ssl": options.get("verify_ssl"),
             }
-
-            registry_cred.inputs = inputs
-            registry_cred.save()
+            for key, value in inputs.items():
+                if registry_cred.get_input(key) != value:
+                    registry_cred.inputs[key] = value
+                    changed = True
+            if changed:
+                registry_cred.save()
+                print("'Default Execution Environment Credential' updated.")
+
+        # Create default globally available Execution Environments
+        for ee in reversed(settings.GLOBAL_JOB_EXECUTION_ENVIRONMENTS):
+            _this_ee, ee_created = ExecutionEnvironment.objects.get_or_create(name=ee["name"], defaults={'image': ee["image"], 'credential': registry_cred})
+            if ee_created:
+                changed = True
+                print(f"'{ee['name']}' Default Execution Environment registered.")
+            elif _this_ee.image != ee["image"] or (registry_cred and _this_ee.credential_id != registry_cred.id):
+                _this_ee.image = ee["image"]
+                _this_ee.credential = registry_cred
+                _this_ee.save()
+                changed = True
+                print(f"'{ee['name']}' Default Execution Environment updated.")
+
+        # Create the control plane execution environment that is used for project updates and system jobs
+        control_plane_ee_image = settings.CONTROL_PLANE_EXECUTION_ENVIRONMENT
+        _cp_ee, cp_created = ExecutionEnvironment.objects.get_or_create(
+            name="Control Plane Execution Environment", defaults={'image': control_plane_ee_image, 'managed_by_tower': True, 'credential': registry_cred}
+        )
+        if cp_created:
             changed = True
-
-            if not created:
-                print('Updated Default Execution Environment Credential')
-
-        for ee in reversed(settings.DEFAULT_EXECUTION_ENVIRONMENTS):
-            _, created = ExecutionEnvironment.objects.update_or_create(
-                name=ee["name"], defaults={"image": ee["image"], "managed_by_tower": True, "credential": registry_cred}
-            )
-
-        if created:
+            print("Control Plane Execution Environment registered.")
+        elif _cp_ee.image != control_plane_ee_image or (registry_cred and _cp_ee.credential_id != registry_cred.id):
```

Comment:
@shanemcd I think we need to think about what it means to re-run the command except w/o any credentials...does that mean you want us to null out any existing credential? I think I need another clause here and above if `registry_cred` is null but `_cp_ee.credential` is not null